jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4301,31 +4306,92 @@ ConstantPool getConstantPool() {
  *
  * @return		the host class of the receiver.
  */
-public Class<?> getNestHost() {
-	return getNestHostImpl();
+@CallerSensitive
+public Class<?> getNestHost() throws SecurityException {
+	if (null == nestHost) {
+		nestHost = getNestHostImpl();
+
+		/* The specification requires that if:
+		 *    - the returned class is not the current class
+		 *    - a security manager is present
+		 *    - the caller's class loader is not the same or an ancestor of the returned class
+		 *    - s.checkPackageAccess() disallows access to the package of the returned class
+		 * then throw a SecurityException.
+		 */
+		if (nestHost != this) {
+			SecurityManager securityManager = System.getSecurityManager();
```

Comment:
shouldn't the security check be always done `(null != nestHost)`? first call to `getNestHost` will set the `nestHost` and perform the security check. for subsequent `getNestHost` calls, `nestHost` will not be `null` and no security check will be performed.

Reply:
Missed that - thanks